services/src/main/scala/cromwell/services/keyvalue/impl/KeyValueServiceActor.scala

```diff
@@ -30,23 +31,13 @@ case class KeyValueServiceActor(serviceConfig: Config, globalConfig: Config) ext
     }
   }
 
-  private def doPut(put: KvPut): Future[KvResponse] = {
-    upsertExecutionInfo(
-      put.pair.key.workflowId,
-      put.pair.key.jobKey,
-      Map(put.pair.key.key -> put.pair.value)
-    ).map(_ => KvPutSuccess(put))
+  private def doPut(put: KvPut): Future[KvResponse] = Future.successful {
```

Comment:
In general I agree w/ the sentiment that @cjllanwarne has here. I get skeeved out by stuff which are AOK now but are taking advantage of technicalities and could easily go horribly wrong in a manner which is easy to escape code review.

At the very least it would be helpful to have a comment in an obvious location stating that if this ever *was* async the whole thing should be changed.